openedx/features/subscriptions/utils.py

```diff
@@ -0,0 +1,26 @@
+from courseware.access_utils import ACCESS_DENIED, ACCESS_GRANTED
+from openedx.features.subscriptions.models import UserSubscription
+
+from student.models import User
+from enrollment.data import get_course_enrollment
+
+
+def track_subscription_enrollment(subscription_id, enrollment):
+    """
+    Add user enrollment to valid subscription.
+    """
+    if subscription_id:
+        username = enrollment.get('user')
+        user = User.objects.get(username=username)
+        valid_user_subscription = UserSubscription.get_valid_subscription(user.id)
+        if valid_user_subscription and valid_user_subscription.subscription_id == subscription_id:
+            valid_user_subscription.course_enrollments.add(enrollment)
+
+def check_user_has_active_subscription(user, course):
+    """
+    """
+    user_subscription = UserSubscription.objects.filter(user=user, course_enrollments__course__id=course.id).first()
```

Comment:
What if the same course in multiple subscriptions? Would the user get all subscriptions?

Reply:
I was a little unsure about this myself. We don't have the renew feature in this part but considering we would have it in future, I think it would be better if we check active for all such subscriptions. let me make this change as well